linux/shim/io.go

```diff
@@ -15,8 +15,9 @@ import (
 )
 
 func copyConsole(ctx context.Context, console console.Console, stdin, stdout, stderr string, wg, cwg *sync.WaitGroup) error {
+	fctx := context.Background()
 	if stdin != "" {
-		in, err := fifo.OpenFifo(ctx, stdin, syscall.O_RDONLY, 0)
+		in, err := fifo.OpenFifo(fctx, stdin, syscall.O_RDONLY, 0)
```

Comment:
And fifos are not horrible in general? You don't have to use a timeout if you can sync it up with the nonblock from the other side and determine from that sync result if context should be canceled or not.